src/utils/lineGraph.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright © 2020 The PES Open Source Team pesos@pes.edu
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// This particularar widget is inspired and borrowed from the implementation of https://github.com/cjbassi/gotop
+package utils
+
+import (
+	"image"
+	"sort"
+
+	drawille "github.com/cjbassi/gotop/src/termui/drawille-go"
+	ui "github.com/gizak/termui/v3"
+)
+
+// LineGraph implements a line graph of data points.
+type LineGraph struct {
+	*ui.Block
+
+	Data   map[string][]float64
+	Labels map[string]string
+
+	HorizontalScale int
+	MaxVal          float64
+
+	LineColors       map[string]ui.Color
+	DefaultLineColor ui.Color
+}
+
+// NewLineGraph creates and returns a lineGraph instance
+func NewLineGraph() *LineGraph {
+	return &LineGraph{
+		Block: ui.NewBlock(),
+
+		Data:   make(map[string][]float64),
+		Labels: make(map[string]string),
+
+		HorizontalScale: 5,
+
+		LineColors: make(map[string]ui.Color),
+	}
+}
+
+func (l *LineGraph) Draw(buf *ui.Buffer) {
```

Comment:
godoc :p

Reply:
This is to just match the default interface. Do you want all of these documented? Because it's documented in the widget section.

Reply:
Ah I see, I suggested the change to maintain uniformity in convention. If not documentation, maybe mention something along the lines of "Draw implemented to match the interface of so and so, for more info see package.Draw"

Reply:
> EDIT: None of them are documented even in termui :P

oof
ok upto you then XD

Reply:
I'm going to leave it be, it's uniform with termui widgets this way